pkg/ingester/ingester.go

```diff
@@ -681,10 +679,9 @@ func (i *Ingester) PushWithCleanup(ctx context.Context, req *mimirpb.WriteReques
 	}
 	defer db.releaseAppendLock()
 
-	span := opentracing.SpanFromContext(ctx)
-	if span != nil {
-		span.LogFields(otlog.String("event", "acquired append lock"))
-	}
+	// Note that we don't .Finish() the span in this method on purpose
+	spanlog := spanlogger.FromContext(ctx, i.logger)
+	level.Debug(spanlog).Log("event", "acquired append lock")
```

Comment:
Given we removed the `if span != nil`, aren't you going to pay the cost of the log formatting even if a request won't be traced?

Reply:
We're logging at debug so it shouldn't actually be formatted in our default config.  Additionally, besides a `Duration` everything else is a string or number which I wouldn't expect to be too expensive.